brave/src/main/java/brave/internal/codec/CharSequences.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.codec;
+
+/**
+ * Most of our parsing tools accept {@link CharSequence} instead of {@link String} to avoid
+ * unnecessary allocation. This contains common functions available on {@link String}, preferring
+ * signatures that match our utilities such as {@link EntrySplitter}.
+ */
+public final class CharSequences {
+
+  /**
+   * Joins two character sequences together. Inputs must be immutable. This is useful when the
+   * result is only scanned. It does not implement {@link #toString()} or {@link #equals(Object)}.
+   */
+  public static CharSequence concat(CharSequence left, CharSequence right) {
```

Comment:
Don't think I see this used in this PR so just curious what situations this would be used for.

Reply:
Good point, I think initially I thought I would use concat for writing a combined header itself (ex "b3=1"+",otherthings"), but noticed StringBuilder does a better job when the result is a string. and string is required for Setter anyway.

The type used inside the method is a part of implementing withoutSubSequence, used to black hole b3's entry in tracestate, but you are right that the static method isn't used. I was also two minds on this, but no problem moving the javadoc down to the inner type and raising it if there were any later uses.

👍